lib/pure/batteries.nim

```diff
@@ -0,0 +1,57 @@
+#
+#
+#            Nim's Runtime Library
+#        (c) Copyright 2012 Andreas Rumpf
+#
+#    See the file "copying.txt", included in this
+#    distribution, for details about the copyright.
+#
+
+when defined(nimdoc) and isMainModule:
+  from std/compileSettings import nil
+  when compileSettings.querySetting(compileSettings.SingleValueSetting.projectFull) == currentSourcePath:
+    ## This is a module that simply imports common modules for your convenience:
+    ##
+    ## .. code-block:: nim
+    ##   import batteries
+    ##
+    ## Same as:
+    ##
+    ## .. code-block:: nim
+    ##   import std/[os, strutils, times, parseutils, hashes, tables, sets,
+    ##     sequtils, parseopt, strformat, sugar, options, strscans, algorithm,
+    ##     math]
+    ##
+    ## This module is similar to the `prelude module <prelude.html>` which it
+    ## deprecates.
+    ## The main differences with prelude are that this module must be imported
+    ## (while prelude had to be included), and that it imports a few more modules
+    ## than prelude did.
+    ## 
+    ## Importing this module never triggers a UnusedImport warning, even if you
+    ## don't use any if the modules it imports.
+    runnableExamples:
+      import std/batteries
+        # same as:
+        # import std/[os, strutils, times, parseutils, hashes, tables, sets,
+        #   sequtils, parseopt, strformat, sugar, options, strscans, algorithm,
+        #   math]
+      let x = 1
+      assert "foo $# $#" % [$x, "bar"] == "foo 1 bar"
```

Comment:
it could be nice to have an assert with a function for each module imported (it would also work as a test that all imports are working, since IIRC runnableExamples are all run) and possibly pick a good representative (now that we have strformat I would not pick `%` as a representative for strutils ;))

Reply:
I've added uses of functions of each of the imported modules.